dissect/cstruct/types/structure.py

```diff
@@ -401,6 +407,46 @@ def __repr__(self) -> str:
 
         return f"<{self.__class__.__name__} {' '.join(values)}>"
 
+    @property
+    def __sizes__(self) -> Mapping[str, int | None]:
+        return ChainMap(self.__class__.__static_sizes__, self.__dynamic_sizes__)
+
+    @property
+    def __values__(self) -> MutableMapping[str, Any]:
+        return StructureValuesProxy(self)
+
+
+class StructureValuesProxy(MutableMapping):
+    """A proxy for the values of fields of a Structure."""
+
+    def __init__(self, struct: Structure):
+        self._struct: Structure = struct
+        # Unfortunately, we need to support code that adds attributes dynamically :/
```

Comment:
We do? To a structure object? Where does such code live?

Reply:
For example [here](https://github.com/fox-it/dissect.etl/blob/3219d88bb6c6668fedb40a5394519c3cef54ee89/dissect/etl/headers/event.py#L51)

Since  ETL has been programmed against an older version of cstruct, I asked MiauwKeru  and  he confirmed that adding attributes should be possible, even if the structure has no corresponding field.

Moreover, we don´t check when setting an attribute to see if it has a corresponding field, so I gathered that this is expected behavior.

Personally, I think adding new attributes this way is is a bad idea. 

Should I remove this capability? 
And perhaps add a check to `setattr` of structure that we don´t create attributes without corresponding fields?